docs/user-guide/lazy/execution.md

```diff
@@ -38,6 +38,10 @@ Above we see that from the 10 million rows there are 14,029 rows that match our
 
 With the default `collect` method Polars processes all of your data as one batch. This means that all the data has to fit into your available memory at the point of peak memory usage in your query.
 
+!!! warning "Reusing `LazyFrame` objects"
+
+    Remember that `LazyFrame`s are query plans i.e. a promise on computation and does not do any implicit caching. This means that every time you reuse it in separate downstream queries after it is defined, it is computed all over again. If you define an operation on a `LazyFrame` that doesn't maintain row order (such as a `groupby`), then the order will also change every time it is run. To avoid this, use `maintain_order=True` arguments for such operations.
```

Comment:
Correction, it is not guaranteed to cache common subplans. We might, but we we don't guarantee it.